go-controller/pkg/ovn/policy_unit_test.go

```diff
@@ -0,0 +1,108 @@
+			errExp:    true,
+			onRetArgMockGoOvnNBClient: []ovntest.TestifyMockHelper{
+				{
+					OnCallMethodName: "ACLAdd", OnCallMethodArgType: []string{"string", "string", "string", "string", "int", "map[string]string", "bool", "string", "string"}, RetArgList: []interface{}{nil, goovn.ErrorExist},
+				},
+			},
+		},
+		{
+			desc:      "test when ACL already exists and confirm no error returned",
+			inpSwName: "testSW",
+			inpMgmtIp: ovntest.MustParseIP("192.168.10.10"),
+			onRetArgMockGoOvnNBClient: []ovntest.TestifyMockHelper{
+				{
+					OnCallMethodName: "ACLAdd", OnCallMethodArgType: []string{"string", "string", "string", "string", "int", "map[string]string", "bool", "string", "string"}, RetArgList: []interface{}{&goovn.OvnCommand{}, nil},
+				},
+				{
+					OnCallMethodName: "Execute", OnCallMethodArgType: []string{"*goovn.OvnCommand"}, RetArgList: []interface{}{goovn.ErrorExist},
+				},
+			},
+		},
+		{
+			desc:      "test when ovnNBClient.Execute() fails",
+			inpSwName: "testSW",
+			inpMgmtIp: ovntest.MustParseIP("192.168.10.10"),
+			errMatch:  fmt.Errorf("failed to create the node acl for logical_switch"),
+			onRetArgMockGoOvnNBClient: []ovntest.TestifyMockHelper{
+				{
+					OnCallMethodName: "ACLAdd", OnCallMethodArgType: []string{"string", "string", "string", "string", "int", "map[string]string", "bool", "string", "string"}, RetArgList: []interface{}{&goovn.OvnCommand{}, nil},
+				},
+				{
+					OnCallMethodName: "Execute", OnCallMethodArgType: []string{"*goovn.OvnCommand"}, RetArgList: []interface{}{goovn.ErrorOption},
+				},
+			},
+		},
+		{
+			desc:      "positive: test ip4 managment ip",
+			inpSwName: "testSW",
+			inpMgmtIp: ovntest.MustParseIP("192.168.10.10"),
+			onRetArgMockGoOvnNBClient: []ovntest.TestifyMockHelper{
+				{
+					OnCallMethodName: "ACLAdd", OnCallMethodArgType: []string{"string", "string", "string", "string", "int", "map[string]string", "bool", "string", "string"}, RetArgList: []interface{}{&goovn.OvnCommand{}, nil},
+				},
+				{
+					OnCallMethodName: "Execute", OnCallMethodArgType: []string{"*goovn.OvnCommand"}, RetArgList: []interface{}{nil},
+				},
+			},
+		},
+		{
+			desc:      "positive: test ip6 management ip",
+			inpSwName: "testSW",
+			inpMgmtIp: ovntest.MustParseIP("fd01::1234"),
+			onRetArgMockGoOvnNBClient: []ovntest.TestifyMockHelper{
+				{
+					OnCallMethodName: "ACLAdd", OnCallMethodArgType: []string{"string", "string", "string", "string", "int", "map[string]string", "bool", "string", "string"}, RetArgList: []interface{}{&goovn.OvnCommand{}, nil},
+				},
+				{
+					OnCallMethodName: "Execute", OnCallMethodArgType: []string{"*goovn.OvnCommand"}, RetArgList: []interface{}{nil},
+				},
+			},
+		},
+	}
+
+	for i, tc := range tests {
+		t.Run(fmt.Sprintf("%d:%s", i, tc.desc), func(t *testing.T) {
+			ovntest.ProcessMockFnList(&mockGoOvnNBClient.Mock, tc.onRetArgMockGoOvnNBClient)
+
+			err := addAllowACLFromNode(tc.inpSwName, tc.inpMgmtIp, mockGoOvnNBClient)
```

Comment:
Since `addAllowACLFromNode` already takes the mock client as argument. It seems like we can follow the example in go_ovn_test.go and then you don't need the changes in go_ovn.go.

Reply:
Good catch, Thanks